googletest_macro/src/lib.rs

```diff
@@ -163,3 +163,108 @@ fn is_test_attribute(attr: &Attribute) -> bool {
             && last_segment.ident == "rstest"
             && attr.path().segments.len() <= 2)
 }
+
+#[proc_macro]
+#[allow(non_snake_case)]
+pub fn TEST_F(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
```

Comment:
Here and below: It's not, to my knowledge, idiomatic to use upper snake case for macros in Rust. This is one respect in which Rust conventions diverges from C++. Perhaps it would be best to stick with the Rust conventions here?

Reply:
That's fair. I thought about this PR as, "how close can this be to C++ GoogleTest?". This led to breaking a few Rust conventions (more `#[allow(non_snake_care)]` but also the `TestFixture` trait definition which is close to `testing::Test` and could deserve a more idiomatic style).

However, changing `TEST_F!(...)` into `test_f!(...)` is better to keep the style of Rust and GoogleTest Rust, without going too far from the original GoogleTest. There is already `expect_that!(...)` and not `EXPECT_THAT!(...)` and so on...

Reply:
Quick update: just changing `TEST` -> `test` is not possible due to name clash with the attribute macro.

Considered solutions:
* Keep `TEST`
* Make `TEST` -> `test` and change the name of `test` to something else
* * We should probably then motivate usage of the C++ GoogleTest style instead of the rust style or support both through the usage of a flag.
* Make `TEST` into another lower case identifier, like `test_s` for test suite.

Reply:
I'd urge against renaming the existing `test` macro. Doing so could break interactions with other libraries like rstest and Tokio. They tend to look for invocations of a macro named `test`, just as we do.

Perhaps renaming this to either `test_case` or `test_suite` would make sense?